redwood-treehouse/src/hostMain/kotlin/app/cash/redwood/treehouse/EventListener.kt

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (C) 2022 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.redwood.treehouse
+
+import app.cash.zipline.Call
+import app.cash.zipline.CallResult
+import app.cash.zipline.ZiplineService
+
+public abstract class EventListener {
+  public open fun appCreated(
+    app: TreehouseApp<*>,
+  ) {
+  }
+
+  public open fun appCanceled(
+    app: TreehouseApp<*>,
+  ) {
+  }
+
+  public open fun codeLoadStart(
+    app: TreehouseApp<*>,
+    manifestUrl: String?,
+  ): Any? = Unit
+
+  public open fun codeLoadEnd(
```

Comment:
Is this called for both the success and fail cases? If it's only for success I might rename the methods to `codeLoadSuccess`/`codeLodeFailure`.